packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/LinkToCategoriesTests.java

```diff
@@ -0,0 +1,276 @@
+
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains("cm:categories");
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(category.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(secondCategory.getId());
+        fileNode.assertThat().field(PROPERTIES_FIELD).contains(thirdCategory.getId());
+    }
+
+    /**
+     * Try to link content to category as an improper user and expect 403 (Forbidden)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_usingUserWithoutAccessRightsAndExpect403()
+    {
+        STEP("Try to link content to a category using user without read permission and expect 403");
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(category.getId());
+        final UserModel userWithoutRights = dataUser.createRandomTestUser();
+        restClient.authenticateUser(userWithoutRights).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+    }
+
+    /**
+     * Try to link content to category using non-existing category and expect 404 (Not Found)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_usingNonExistingCategoryAndExpect404()
+    {
+        STEP("Try to link content to non-existing category and expect 404");
+        final String nonExistingCategoryId = "non-existing-dummy-id";
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(nonExistingCategoryId);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Try to link content to category passing empty list as input and expect 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_passingEmptyListAndExpect400()
+    {
+        STEP("Try to call link content API with empty list and expect 400");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategories(Collections.emptyList());
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to link content to category passing invalid ID in input list and expect 400 (Bad Request)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testLinkContentToCategory_passingEmptyIdAndExpect400()
+    {
+        STEP("Try to call link content API with empty category ID and expect 400");
+        final String nonExistingCategoryId = StringUtils.EMPTY;
+        final RestCategoryLinkBodyModel categoryLink = createCategoryLinkWithId(nonExistingCategoryId);
+        restClient.authenticateUser(user).withCoreAPI().usingNode(file).linkToCategory(categoryLink);
+
+        restClient.assertStatusCodeIs(BAD_REQUEST);
+    }
+
+    /**
+     * Try to link non-content node to category and expect 400 (Bad Request)
```

Comment:
I'm surprised at this. I thought that Share allowed folders to be linked to categories.

Reply:
Yes - it's probably a good idea to look to see if there's any restriction in the type of node in the existing v0 webscript.

Reply:
I've just verified that and you are true. Share in did allows folder linking. I will fix that.

Reply:
Regarding other node types - v0 API allows to link different kinds of nodes to categories. I don't think there's any check for that. I've asked Stefan if we should support only Content and Folder nodes.

Reply:
It's a good idea to check. For now I would take the check out completely so that this can be merged. Then we can have a new ticket for the validation once we know what the requirements are.